pkg/quick/quick_test.go

```diff
@@ -19,77 +19,95 @@
 package quick
 
 import (
+	"bytes"
 	"encoding/json"
 	"io/ioutil"
 	"os"
 	"reflect"
 	"runtime"
 	"strings"
 	"testing"
-
-	. "gopkg.in/check.v1"
 )
 
-func Test(t *testing.T) { TestingT(t) }
-
-type MySuite struct{}
-
-var _ = Suite(&MySuite{})
-
-func (s *MySuite) TestReadVersion(c *C) {
+func TestReadVersion(t *testing.T) {
 	type myStruct struct {
 		Version string
 	}
 	saveMe := myStruct{"1"}
 	config, err := New(&saveMe)
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
 	err = config.Save("test.json")
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	version, err := GetVersion("test.json")
-	c.Assert(err, IsNil)
-	c.Assert(version, Equals, "1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if version != "1" {
+		t.Fatalf("Expected version '1', got '%v'", version)
+	}
 }
 
-func (s *MySuite) TestReadVersionErr(c *C) {
+func TestReadVersionErr(t *testing.T) {
 	type myStruct struct {
 		Version int
 	}
 	saveMe := myStruct{1}
 	_, err := New(&saveMe)
-	c.Assert(err, Not(IsNil))
+	if err == nil {
+		t.Fatal("Unexpected should have failed")
+	}
 
 	err = ioutil.WriteFile("test.json", []byte("{ \"version\":2,"), 0644)
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	_, err = GetVersion("test.json")
-	c.Assert(err, Not(IsNil))
+	if err == nil {
```

Comment:
Log can have more information. That way, the user knows where exactly this failure happened and why.

Reply:
Fixed